docs/architecture/overview.md

```diff
@@ -37,6 +45,25 @@ This document specifies the architecture for GCP Ingestion as a whole.
     - Expected initial topics are Structured Ingestion, Telemetry, and Pioneer
 - Must accept configuration defining HTTP headers to capture
 
+### Account Ecosystem Decryptor
+
+- Must have access restricted to a limited set of operators
+- Must have access to a set of encrypted private keys stored in
+  the source repository or on GCS
+- Must have access to KMS to decrypt the private keys on startup,
+  holding them in memory
+- Must throw away any payload that does not match expected structure
+  - The payload must contain valid JSON
+  - The payload may be compressed via `gzip`
+  - The payload must contain a top-level field called `ecosystem_anon_id`
+    and must _not_ contain top-level fields `ecosystem_user_id` or
+    `previous_ecosystem_user_ids` (which indicate a client not conforming
+    to Account Ecosystem Telemetry contracts)
+- Must replace field `ecosystem_anon_id` with the decrypted value in a
+  field called `ecosystem_user_id`
+- Must replace array-type field `previous_ecosystem_anon_ids` if it exists
+  with the decrypted values in a field called `previous_ecosystem_user_ids`
```

Comment:
This `previous_ecosystem_anon_ids` array gives us the flexibility to implement either a "ID changed ping" where the array would have a single value, or a deletion request ping with the whole history of values that fxa knows about.

Reply:
I like this FWIW, feels like it gives us some good optionality